SCRIVI PRESENTAZIONE PROGETTO

In [8]:
#import libraries
import numpy as np
import pandas as pd

## Section 1 -Loading and dataset exploration-

## Market Temperature

### Loading market temperature dataset

In [9]:
market_temp=pd.read_csv("Market_temp_index_uc_month.csv")
market_temp

RegionID  SizeRank       RegionName RegionType StateName  2018-01-31  \
0      102001         0    United States    country       NaN        50.0   
1      394913         1     New York, NY        msa        NY        53.0   
2      753899         2  Los Angeles, CA        msa        CA        69.0   
3      394463         3      Chicago, IL        msa        IL        48.0   
4      394514         4       Dallas, TX        msa        TX        55.0   
..        ...       ...              ...        ...       ...         ...   
923    753929       935       Zapata, TX        msa        TX         NaN   
924    394743       936    Ketchikan, AK        msa        AK        37.0   
925    753874       937        Craig, CO        msa        CO         3.0   
926    395188       938       Vernon, TX        msa        TX        51.0   
927    394767       939       Lamesa, TX        msa        TX        46.0   

     2018-02-28  2018-03-31  2018-04-30  2018-05-31  ...  2024-02-29  \
0          50.0        52.0        54.0        55.0  ...        63.0   
1          52.0        55.0        57.0        56.0  ...        92.0   
2          66.0        66.0        67.0        66.0  ...        85.0   
3          49.0        51.0        52.0        51.0  ...        74.0   
4          55.0        58.0        60.0        59.0  ...        67.0   
..          ...         ...         ...         ...  ...         ...   
923         NaN         NaN         NaN         NaN  ...        39.0   
924        35.0        34.0        37.0        49.0  ...        74.0   
925        23.0        52.0        67.0        56.0  ...        61.0   
926        45.0         NaN        48.0        65.0  ...        83.0   
927        45.0        43.0        49.0        50.0  ...        47.0   

     2024-03-31  2024-04-30  2024-05-31  2024-06-30  2024-07-31  2024-08-31  \
0          63.0        62.0        59.0        57.0        55.0        53.0   
1          90.0        87.0        81.0        78.0        77.0        77.0   
2          83.0        80.0        74.0        71.0        67.0        65.0   
3          76.0        75.0        71.0        67.0        64.0        61.0   
4          67.0        64.0        59.0        55.0        52.0        49.0   
..          ...         ...         ...         ...         ...         ...   
923        36.0        27.0        40.0        53.0        47.0        31.0   
924        69.0        49.0        53.0        61.0        66.0        57.0   
925        55.0        59.0        57.0        48.0        33.0        21.0   
926        78.0        64.0        65.0        61.0        52.0        57.0   
927        44.0        47.0        48.0        41.0        37.0        36.0   

     2024-09-30  2024-10-31  2024-11-30  
0          52.0        51.0        50.0  
1          74.0        71.0        71.0  
2          63.0        62.0        63.0  
3          58.0        56.0        56.0  
4          48.0        49.0        48.0  
..          ...         ...         ...  
923        34.0        42.0        41.0  
924        53.0        44.0        50.0  
925        28.0        41.0        37.0  
926        56.0        51.0        43.0  
927        39.0        39.0        40.0  

[928 rows x 88 columns]

In [10]:
market_temp.shape

(928, 88)

Once the dataset has been load, we notice that 83 columns (column 5-87) are dates. In order to perform further time series analysis it is necessary to have these columns as indexes.

In [11]:
# Trasforma il dataset usando melt
market_temp = market_temp.melt(
    id_vars=['RegionID', 'SizeRank', 'RegionName', 'RegionType', 'StateName'],  # Colonne che rimangono fisse
    var_name='Date',  # Nome per la nuova colonna con le date
    value_name='MarketTemperature'  # Nome per i valori delle temperature
)

# Converte la colonna 'Date' in formato datetime
market_temp['Date'] = pd.to_datetime(market_temp['Date'])

# Imposta l'indice temporale (opzionale: mantieni RegionID come chiave)
market_temp_melted = market_temp.set_index(['Date'])

# Visualizza il risultato
market_temp_melted.head(50)
market_temp_melted.index

DatetimeIndex(['2018-01-31', '2018-01-31', '2018-01-31', '2018-01-31',
               '2018-01-31', '2018-01-31', '2018-01-31', '2018-01-31',
               '2018-01-31', '2018-01-31',
               ...
               '2024-11-30', '2024-11-30', '2024-11-30', '2024-11-30',
               '2024-11-30', '2024-11-30', '2024-11-30', '2024-11-30',
               '2024-11-30', '2024-11-30'],
              dtype='datetime64[ns]', name='Date', length=77024, freq=None)

In [12]:
market_temp_melted.index.unique()

DatetimeIndex(['2018-01-31', '2018-02-28', '2018-03-31', '2018-04-30',
               '2018-05-31', '2018-06-30', '2018-07-31', '2018-08-31',
               '2018-09-30', '2018-10-31', '2018-11-30', '2018-12-31',
               '2019-01-31', '2019-02-28', '2019-03-31', '2019-04-30',
               '2019-05-31', '2019-06-30', '2019-07-31', '2019-08-31',
               '2019-09-30', '2019-10-31', '2019-11-30', '2019-12-31',
               '2020-01-31', '2020-02-29', '2020-03-31', '2020-04-30',
               '2020-05-31', '2020-06-30', '2020-07-31', '2020-08-31',
               '2020-09-30', '2020-10-31', '2020-11-30', '2020-12-31',
               '2021-01-31', '2021-02-28', '2021-03-31', '2021-04-30',
               '2021-05-31', '2021-06-30', '2021-07-31', '2021-08-31',
               '2021-09-30', '2021-10-31', '2021-11-30', '2021-12-31',
               '2022-01-31', '2022-02-28', '2022-03-31', '2022-04-30',
               '2022-05-31', '2022-06-30', '2022-07-31', '2022-08-31',
      

Now the datset has a correct structure: for each RegionID, which are 928 overall, we have all the 83 recordings about the market temperature. We expect 928 * 83 = 77024 rows.

In [13]:
market_temp_melted.shape

(77024, 6)

### Columns exploration

In [14]:
#regionid exploration
print(len(market_temp['RegionID'].unique()))

928


In [15]:
#size rank exploration
print(len(market_temp['SizeRank'].unique()))

925


In [16]:
#region name exploration, we expect the same number of RegionID
print(len(market_temp['RegionName'].unique()))

928


In [17]:
#region type exploration
print(market_temp['RegionType'].unique())
print(f"the number of region types is: {len(market_temp['RegionType'].unique())}")

['country' 'msa']
the number of region types is: 2


In [18]:
#state name exploration
print(market_temp['StateName'].unique())
print(f"the number of region types is: {len(market_temp['StateName'].unique())}")

[nan 'NY' 'CA' 'IL' 'TX' 'VA' 'PA' 'FL' 'GA' 'MA' 'AZ' 'MI' 'WA' 'MN' 'CO'
 'MD' 'MO' 'NC' 'OR' 'OH' 'NV' 'IN' 'TN' 'RI' 'WI' 'OK' 'KY' 'LA' 'UT'
 'CT' 'AL' 'HI' 'NE' 'SC' 'NM' 'ID' 'AR' 'IA' 'KS' 'MS' 'ME' 'NH' 'DE'
 'AK' 'NJ' 'SD' 'WV' 'ND' 'VT' 'MT' 'WY']
the number of region types is: 51


In [19]:
#we detect a nan state, which values are associated to it?
market_temp[market_temp['StateName'].isna()]


RegionID  SizeRank     RegionName RegionType StateName       Date  \
0        102001         0  United States    country       NaN 2018-01-31   
928      102001         0  United States    country       NaN 2018-02-28   
1856     102001         0  United States    country       NaN 2018-03-31   
2784     102001         0  United States    country       NaN 2018-04-30   
3712     102001         0  United States    country       NaN 2018-05-31   
...         ...       ...            ...        ...       ...        ...   
72384    102001         0  United States    country       NaN 2024-07-31   
73312    102001         0  United States    country       NaN 2024-08-31   
74240    102001         0  United States    country       NaN 2024-09-30   
75168    102001         0  United States    country       NaN 2024-10-31   
76096    102001         0  United States    country       NaN 2024-11-30   

       MarketTemperature  
0                   50.0  
928                 50.0  
1856                52.0  
2784                54.0  
3712                55.0  
...                  ...  
72384               55.0  
73312               53.0  
74240               52.0  
75168               51.0  
76096               50.0  

[83 rows x 7 columns]

When the Region Name is United States, the data point records the value for the all country, not just for a specific metropolitan area. It is probably an average value for the all nation.

The region id is a unique identifier of the house properties

## Median Sales price

### Loading median sales price dataset

In [20]:
median_sales_price=pd.read_csv('Median_sale_price_uc_month.csv')
median_sales_price

RegionID  SizeRank       RegionName RegionType StateName  2008-02-29  \
0      102001         0    United States    country       NaN    170922.0   
1      394913         1     New York, NY        msa        NY    397000.0   
2      753899         2  Los Angeles, CA        msa        CA    470000.0   
3      394463         3      Chicago, IL        msa        IL    216750.0   
4      394514         4       Dallas, TX        msa        TX    138000.0   
..        ...       ...              ...        ...       ...         ...   
708    394869       869      Moberly, MO        msa        MO         NaN   
709    394371       891     Beatrice, NE        msa        NE         NaN   
710    753914       908  Port Lavaca, TX        msa        TX         NaN   
711    395003       912        Price, UT        msa        UT         NaN   
712    394805       916   Los Alamos, NM        msa        NM         NaN   

     2008-03-31  2008-04-30  2008-05-31  2008-06-30  ...  2024-01-31  \
0      175612.0    177500.0    180000.0    185000.0  ...    324000.0   
1      390000.0    390000.0    390000.0    399900.0  ...    570000.0   
2      455000.0    457000.0    440000.0    435000.0  ...    865000.0   
3      220000.0    221000.0    227000.0    235000.0  ...    279000.0   
4      146000.0    144950.0    150000.0    156000.0  ...    368000.0   
..          ...         ...         ...         ...  ...         ...   
708         NaN         NaN         NaN         NaN  ...         NaN   
709         NaN         NaN         NaN         NaN  ...         NaN   
710         NaN         NaN         NaN         NaN  ...         NaN   
711         NaN         NaN         NaN         NaN  ...         NaN   
712         NaN         NaN         NaN         NaN  ...    584600.0   

     2024-02-29  2024-03-31  2024-04-30  2024-05-31  2024-06-30  2024-07-31  \
0      335000.0    345000.0    350000.0    360000.0    369000.0    361000.0   
1      565000.0    575000.0    585000.0    615000.0    640000.0    649450.0   
2      915000.0    916250.0    950000.0    970000.0    960000.0    950500.0   
3      285000.0    300000.0    315000.0    325000.0    337275.0    330000.0   
4      380000.0    390000.0    400000.0    405000.0    400000.0    395000.0   
..          ...         ...         ...         ...         ...         ...   
708         NaN         NaN    139000.0    152000.0    163250.0    180500.0   
709         NaN         NaN         NaN         NaN         NaN         NaN   
710         NaN         NaN         NaN         NaN         NaN    177884.0   
711         NaN         NaN         NaN         NaN         NaN         NaN   
712    589000.0    665000.0    542500.0    605017.0    538634.0    529461.0   

     2024-08-31  2024-09-30  2024-10-31  
0      360000.0    350500.0    355000.0  
1      640000.0    625000.0    624451.0  
2      932000.0    915000.0    941000.0  
3      320000.0    317881.0    315000.0  
4      390000.0    380000.0    384000.0  
..          ...         ...         ...  
708    186000.0     46338.0    185900.0  
709         NaN         NaN    200000.0  
710    209255.0    103075.0    140948.0  
711         NaN    228500.0    305000.0  
712    535000.0    552500.0    635000.0  

[713 rows x 206 columns]

### Columns exploration